### ISSUE TITLE
[BUG] Playwright traces for Microsoft Teams tests using webview2 desktop client do not include css stylesheets

### ISSUE THREAD

**Issue by ericcjobes** [microsoft/playwright#27026]
### System info
- Playwright Version: 1.35, 1.37. Repros on both `trace.playwright.dev` and `trace.playwright.dev/next`
- Operating System: trace generated on Windows 11 running Webview2 desktop Teams client, viewed on any OS
- Browser: trace viewed on any browser

### Source code

- [ x ] I cannot provide the source code to repro as it is hosted in confidential Azure DevOps repository. However, I can provide an example playwright trace over Teams/Email to internal Microsoft developers upon request.

**Link to the GitHub repository with the repro**
N/A, Teams source code is hosted in private Azure DevOps repository. Happy to provide link over Teams if you would like to look into our setup.

**Config file**

```js
// playwright.config.ts
import { defineConfig, PlaywrightTestConfig } from '@playwright/test';
import { ITeamsTestConfiguration } from "./common/configurations/test-configuration.interface"; // can provide details of this upon request

const isServerBuild = () => !!process.env.TF_BUILD;
/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
import { config as dotenvConfig } from "dotenv";

const config: PlaywrightTestConfig<{
  projectConfig: ITeamsTestConfiguration;
}> = {
  // testDir: "./tests",
  /* Maximum time one test can run for. */
  timeout: defaultTestTimeout,
  expect: {
    /**
     * Maximum time expect() should wait for the condition to be met.
     * For example in `await expect(locator).toHaveText();`
     */
    timeout: 5000,
  },
  /* Run tests in files in parallel */
  fullyParallel: false,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? defaultRetriesInCI : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: [
    [
      "html",
      {
        outputFolder: `playwright-report/reportTests`,
        open: isServerBuild() ? "never" : "on-failure",
      },
    ],
    ["junit", { outputFile: "playwright-report/reportTests/results.xml" }],
    // removed internal reporter used for Teams
    ["list"],
  ],
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    /* Maximum time each action such as `click()` can take. Defaults to 0 (no limit). */
    actionTimeout: 0,
    /* Base URL to use in actions like `await page.goto('/')`. */
    // baseURL: 'http://localhost:3000',

    /* Collect trace on all tests for now to get data See https://playwright.dev/docs/trace-viewer */
    trace: "on",
    video: "on",
    testIdAttribute: "data-tid",
  },
  globalSetup: "./setup/global-setup",
  globalTeardown: "./setup/global-teardown",

  /* Configure projects for major browsers */
  projects: [
    // removed sensitive info
  ]
};

export default defineConfig(config);
```

**Test file (self-contained)**
Note: due to our involved setup, these cannot be run stand-alone as they require our supporting infrastructure
```js
test("should launch the app", async () => {
    await teamsApp.launch({
      glassJarSeed: new T1Chat(),
      skipLogin: true,
    });
  });

  test("should enter chat app", async () => {
    await teamsApp.launch({
      glassJarSeed: new T1Chat(),
      skipLogin: false, // Temporarily set skipLogin to false until config refactor
    });
    await teamsApp.Apps.chatApp.goToChatApp();
  });
```

**Steps**
- [Run the test against teams desktop application]
- [Open the generated playwright trace]

**Expected**

- Playwright trace includes necessary .css files to render page and styles properly
- following screenshot taken from browser test
- we believe this works because teams.microsoft.com has the css defined as remote sources
![image](https://github.com/microsoft/playwright/assets/56457218/c09598f3-e7f9-4a46-b9c0-fd0fe0942043)

**Actual**
- When run against desktop test, playwright trace do not contain necessary .css files to render page properly, making debugging inside the snapshot-container challenging

![image](https://github.com/microsoft/playwright/assets/56457218/3553b9e7-153d-48be-b7ee-6c6402b1cb79)
- in playwright trace console, we see the following errors, which are not present for the traces against web-only which do not reproduce the issue:
![image](https://github.com/microsoft/playwright/assets/56457218/af84ff6c-e754-4d3d-9f99-3c5e42764886)
- the failing endpoint is pointing to a css file which can be either locally hosted or on CDN (both situations repro), which would be present when tests are running and trace is recorded, but not after the fact when the trace is viewed. example url format: `https://TeamsEndpointUrl>/hashed-assets/griffel-<guid>.css`.

Is there anything we can change in our config to ensure the .css files, which are available during the test, are then bundled with the playwright trace, such that when just opening the trace without identical local server running, we can load stylesheets correctly?

It is worth noting that the stylesheets are rendered properly during the tests, as the screenshots in the playwright timeline render the application correctly, as do our ffmpeg-recorded videos.

**Comment by ericcjobes** [comment]
Thank you for the suggestion. I'm testing a small refactor to move our `tracing.start()` calls to be as early in our startup process as possible. Will let you know the results!

**Comment by aslushnikov** [comment]
Closing this since the discussion has stalled. Please file a new issue if you think we still can be helpful!

**Comment by ericcjobes** [comment]
Following up - the fix proposed did work. While it wasn't as simple as above suggestion since we have complicated app startup a bit more, generally we were starting the trace after launching our product. To fix, I ensured trace was started as soon as browserContext could be initialized, even before we navigate to teams.microsoft.com (for web) or load teams into our webview (for desktop). Thank you for the help!
